Mathlib/Probability/Moments/IntegrableExpMul.lean

```diff
@@ -0,0 +1,452 @@
+then `X ^ n * exp (v * X)` is integrable for all `n : ℕ`. -/
+lemma integrable_pow_mul_exp_of_mem_interior_integrableExpSet
+    (hv : v ∈ interior (integrableExpSet X μ)) (n : ℕ) :
+    Integrable (fun ω ↦ X ω ^ n * exp (v * X ω)) μ := by
+  rw [← integrable_norm_iff]
+  · simp_rw [norm_eq_abs, abs_mul, abs_pow, abs_exp]
+    exact integrable_pow_abs_mul_exp_of_mem_interior_integrableExpSet hv n
+  · have hX : AEMeasurable X μ := aemeasurable_of_mem_interior_integrableExpSet hv
+    exact ((hX.pow_const _).mul
+      (measurable_exp.comp_aemeasurable (hX.const_mul _))).aestronglyMeasurable
+
+/-- If 0 belongs to the interior of the interval `integrableExpSet X μ`,
+then `|X| ^ n` is integrable for all `n : ℕ`. -/
+lemma integrable_pow_abs_of_mem_interior_integrableExpSet
+    (h : 0 ∈ interior (integrableExpSet X μ)) (n : ℕ) :
+    Integrable (fun ω ↦ |X ω| ^ n) μ := by
+  convert integrable_pow_abs_mul_exp_of_mem_interior_integrableExpSet h n
+  simp
+
+/-- If 0 belongs to the interior of the interval `integrableExpSet X μ`,
+then `X ^ n` is integrable for all `n : ℕ`. -/
+lemma integrable_pow_of_mem_interior_integrableExpSet
+    (h : 0 ∈ interior (integrableExpSet X μ)) (n : ℕ) :
+    Integrable (fun ω ↦ X ω ^ n) μ := by
+  convert integrable_pow_mul_exp_of_mem_interior_integrableExpSet h n
+  simp
+
+-- todo: move to L1Space
+lemma integrable_norm_rpow_iff {α E : Type*} {_ : MeasurableSpace α} {μ : Measure α}
+    [NormedAddCommGroup E] {p : ℝ≥0∞} {f : α → E}
+    (hf : AEStronglyMeasurable f μ) (p_zero : p ≠ 0) (p_top : p ≠ ∞) :
+    Integrable (fun x : α => ‖f x‖ ^ p.toReal) μ ↔ Memℒp f p μ := by
+  rw [← memℒp_norm_rpow_iff (q := p) hf p_zero p_top, ← memℒp_one_iff_integrable,
+    ENNReal.div_self p_zero p_top]
+
+-- todo: move somewhere?
+lemma rpow_le_add_pow_floor_pow_ceil {x p : ℝ} (hx_nonneg : 0 ≤ x) (hp : 1 ≤ p) :
+    x ^ p ≤ x ^ ⌊p⌋₊ + x ^ ⌈p⌉₊ := by
+  have hn : 1 ≤ ⌊p⌋₊ := Nat.le_floor (mod_cast hp)
+  have hm : 1 ≤ ⌈p⌉₊ := hn.trans (Nat.floor_le_ceil _)
+  by_cases hx_zero : x = 0
+  · rw [hx_zero, zero_pow, zero_pow, zero_rpow, zero_add]
+    · suffices 0 < p by simp [this.ne']
+      positivity
+    · omega
+    · omega
+  rcases le_total x 1 with hx | hx
+  · calc x ^ p
+    _ ≤ x ^ ⌊p⌋₊ := by
+      rw [← rpow_natCast]
+      exact rpow_le_rpow_of_exponent_ge (lt_of_le_of_ne' hx_nonneg hx_zero) hx
+        (Nat.floor_le (by positivity))
+    _ ≤ _ := le_add_of_nonneg_right (by positivity)
+  · calc x ^ p
+    _ ≤ x ^ ⌈p⌉₊ := by
+      rw [← rpow_natCast]
+      exact rpow_le_rpow_of_exponent_le hx (Nat.le_ceil _)
+    _ ≤ _ := le_add_of_nonneg_left (by positivity)
+
+/-- If 0 belongs to the interior of `integrableExpSet X μ`, then `X` is in `ℒp` for all
+finite `p` with `1 ≤ p`.
+If `μ` is a finite measure, the condition `1 ≤ p` is not necessary: see
+`memℒp_of_mem_interior_integrableExpSet`. -/
+lemma memℒp_of_mem_interior_integrableExpSet_of_one_le {p : ℝ≥0} (hp : 1 ≤ p)
```

Comment:
I'm not sure you need `1 ≤ p` here. Quick proof sketch: one has x \le e^x for all real x. So c x \le e ^ {c x}, i.e., x \le c^{-1} e^ {c x}. Applying this to x and -x gives |x| \le c^{-1} max (e^{c x}, e^{-c x}), and then |x|^p \le c^{-p} max (e^{p c x}, e^{-p c x}). Then for any p (be it larger or smaller than 1), you can choose c small enough so that e^{p c X} and e^{-p c X} are integrable, and then you deduce that |X|^p is integrable. This has the advantage to avoid the dance with integer parts also...